include/circt/Dialect/FIRRTL/FIRRTLOpInterfaces.td

```diff
@@ -205,7 +206,11 @@ def FModuleLike : OpInterface<"FModuleLike"> {
       }
       assert(symbols.size() == $_op.getNumPorts());
       symbols[portIndex] = InnerSymAttr::get(symbol);
-      $_op.setPortSymbols(symbols);
+
+      FModuleLike::fixupPortSymsArray(symbols, $_op.getContext());
```

Comment:
I am guessing we don't usually use this to clear out symbols, but if this function returns the array of new ports, then we don't have to copy an array of null/empty port symbols just to call `syms.clear()`. E.g. maybe something like this:
```c++
      if (llvm::none_of(syms, [&](auto &sym) { return sym; })) {
        return ArrayAttr::get(context);
       }
       SmallVector<Attribute> newPorts;
       llvm::transform(ports, back_inserter(newPorts0, [&empty](Attribute attr) { if (attr) return attr; return empty; }
       return ArrayAttr::get(context, newPorts);
    }
```

Reply:
Good call, it'd be good to streamline this.  For now, I'll land this for the perf fix but we should fix/discuss.